File: jambeth/jambeth-service-rest/src/main/java/de/osthus/ambeth/rest/RESTClientServiceFactory.java
package de.osthus.ambeth.rest;

import de.osthus.ambeth.ioc.IInitializingBean;
import de.osthus.ambeth.ioc.factory.IBeanContextFactory;
import de.osthus.ambeth.service.IClientServiceFactory;
import de.osthus.ambeth.service.IOfflineListenerExtendable;

public class RESTClientServiceFactory implements IClientServiceFactory, IInitializingBean
{
	@Override
	public void afterPropertiesSet()
	{
		// Intended blank
	}

	@Override
	public Class<?> getSyncInterceptorType(Class<?> clientInterface)
	{
		return null;
	}

	@Override
	public Class<?> getTargetProviderType(Class<?> clientInterface)
	{
		return RESTClientInterceptor.class;
	}

	@Override
	public String getServiceName(Class<?> clientInterface)
	{
		String name = clientInterface.getSimpleName();
		if (name.endsWith("Client"))
		{
			name = name.substring(0, name.length() - 6) + "Service";
		}
		else if (name.endsWith("WCF"))
		{
			name = name.substring(0, name.length() - 3);
		}
		if (name.startsWith("I"))
		{
			return name.substring(1);
		}
		return name;
	}

	@Override
	public void postProcessTargetProviderBean(String targetProviderBeanName, IBeanContextFactory beanContextFactory)
	{
		beanContextFactory.link(targetProviderBeanName).to(IOfflineListenerExtendable.class);
	}
}

File: ambeth/Ambeth.Service.REST/ambeth/rest/RESTClientServiceFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using De.Osthus.Ambeth.Ioc;
using De.Osthus.Ambeth.Proxy;
using De.Osthus.Ambeth.Service;
using De.Osthus.Ambeth.Ioc.Factory;

namespace De.Osthus.Ambeth.Rest
{
    public class RESTClientServiceFactory : IClientServiceFactory, IInitializingBean
    {
        public virtual void AfterPropertiesSet()
        {
            // Intended blank
        }

        public virtual Type GetSyncInterceptorType(Type clientInterface)
        {
            return null;
        }

        public virtual Type GetTargetProviderType(Type clientInterface)
        {
            return typeof(RESTClientInterceptor);
        }

        public virtual String GetServiceName(Type clientInterface)
        {
            String name = clientInterface.Name;
            if (name.EndsWith("Client"))
            {
                name = name.Substring(0, name.Length - 6) + "Service";
            }
            else if (name.EndsWith("WCF"))
            {
                name = name.Substring(0, name.Length - 3);
            }
            if (name.StartsWith("I"))
            {
                return name.Substring(1);
            }
            return name;
        }


        public virtual void PostProcessTargetProviderBean(String targetProviderBeanName, IBeanContextFactory beanContextFactory)
        {
            beanContextFactory.Link(targetProviderBeanName).To<IOfflineListenerExtendable>();
        }
    }
}
